### PR TITLE
Switch to coverlet.collector

### PR DESCRIPTION
**Pull request by sungam3r**
fixes (probably) #2982

### CODE REVIEW

**Review by Shane32 — APPROVED** [review]

**Comment by Shane32** [comment]
> fixes (probably) #2982

Fixes the coverlet problems anyway. I think I have a solution for some of the issues that have to do with static variables but I have not worked on it yet.

**Comment by sungam3r** [comment]
@Shane32 I think that the reason for increasing the coverage (above) is that files for all TFMs have become transmitted to codecov (only net6 before).
изображение

**Comment by sungam3r** [comment]
Or it is just a difference of coverage calculation between coverlet.msbuild and coverlet.collector.

File: .github/workflows/test-code.yml [hunk 1]
```diff
@@ -61,12 +61,19 @@ jobs:
...
+          --results-directory .coverage
+          --
+          DataCollectionRunSettings.DataCollectors.DataCollector.Configuration.Format=opencover
+          DataCollectionRunSettings.DataCollectors.DataCollector.Configuration.Exclude=[GraphQL.StarWars]*,[GraphQL.Harness]*,[GraphQL.Dummy]*
       - name: Upload coverage to codecov
         if: ${{ startsWith(matrix.os, 'ubuntu') }}
         uses: codecov/codecov-action@v2.1.0
         with:
-          files: .coverage/GraphQL.DataLoader.Tests/coverage.net6.opencover.xml,.coverage/GraphQL.Harness.Tests/coverage.net6.opencover.xml,.coverage/GraphQL.Tests/coverage.net6.opencover.xml,.coverage/GraphQL.MicrosoftDI.Tests/coverage.net6.opencover.xml
+          files: 'src/.coverage/**/coverage.opencover.xml'
```
Comment (sungam3r) [inline .github/workflows/test-code.yml:76]: No need anymore to remember adding new projects here.

Comment (sungam3r) [reply .github/workflows/test-code.yml:76]: I think codecov had no globbing support before.

File: src/Directory.Build.targets [hunk 2]
```diff
@@ -1,11 +1,5 @@
 <Project>
 
-  <PropertyGroup Condition="'$(IsTestProject)' == 'true'">
```
Comment (sungam3r) [inline src/Directory.Build.targets:3]: Moved into `dotnet test` arguments.

**Pull request merged**